packages/features/users/components/UserTable/DeleteMemberModal.tsx

```diff
@@ -13,15 +13,33 @@ export function DeleteMemberModal({ state, dispatch }: { state: State; dispatch:
   const utils = trpc.useUtils();
   const removeMemberMutation = trpc.viewer.teams.removeMember.useMutation({
     onSuccess() {
-      // We don't need to wait for invalidate to finish
-      Promise.all([
-        utils.viewer.teams.get.invalidate(),
-        utils.viewer.eventTypes.invalidate(),
-        utils.viewer.organizations.listMembers.invalidate(),
-      ]);
+      // Update the infinite data to remove the deleted user
+      utils.viewer.organizations.listMembers.setInfiniteData({ limit: 10, searchTerm: "" }, (oldData) => {
+        if (!oldData) return oldData;
+        return {
+          ...oldData,
+          pages: oldData.pages.map((page) => ({
+            ...page,
+            rows: page.rows.filter((member) => member.id !== state.deleteMember.user?.id),
```

Comment:
Stacking array methods makes me question things but I can't think of any alternatives to this.

Reply:
Only way to do this optimistically :/